snake/snake_game_khr.py

```diff
@@ -0,0 +1,165 @@
+from time import sleep
+import os
+import random
+from random import randint
+
+
+class Board:
+    def __init__(self, width, height, top_border="<", bottom_border=">", vert_border="^"):
+        self.snake = Snake()
+        self.top_boarders = top_border
+        self.bottom_boarders = bottom_border
+        self.vert_boarders = vert_border
+        self.width = width
+        self.height = height
+        self.board = self.init_board()
+
+    def init_board(self):
+        new_board = []
+        for i in range(self.height):
+            if i == 0:
+                row = [self.top_boarders] * self.width
+                new_board.append(row)
+            elif i in {self.height - 1}:
+                row = [self.bottom_boarders] * self.width
+                new_board.append(row)
+            else:
+                row = []
+                for j in range(self.width):
+                    if j in {0, self.width - 1}:
+                        row.append(self.vert_boarders)
+                    else:
+                        row.append(" ")
+                new_board.append(row)
+        return new_board
+
+    def show(self):
+        for row in self.board:
+            print(' '.join(row))
+
+
+class Snake:
+    def __init__(self, symbol="o", position=(1, 1)):
+        self.symbol = symbol
+        self.body = [position]
+
+    def eat(self, position: tuple):
+        self.body.append(position)
+
+    def move(self, position: tuple):
+        self.body.append(position)
+        self.body.pop(0)
+
+    def choices(self):
+        next_positions = [
+            (1, 0),
+            (0, 1),
+            (-1, 0),
+            (0, -1)
+        ]
+        result = []
+        for next_position in next_positions:
+            new_position = self.body[-1][0] + next_position[0], self.body[-1][1] + next_position[1]
+            result.append(new_position)
+        return result
+
+
+class Apple:
+    def __init__(self, symbol='$', position=(2, 2)):
+        self.position = position
+        self.symbol = symbol
+
+
+class GameOverError(Exception):
+    pass
+
+
+class Game:
+    def __init__(self, width, height):
+        self.symbol = "o"
+        self.width = width
+        self.height = height
+        self.board = Board(self.width, self.height)
+        self.snake = Snake()
+        self.apple = Apple()
+
+    def init_apple(self):
+        last_pos = self.apple.position
+        self.board = Board(self.width, self.height)
+        positions = []
+        x = randint(1, self.board.height-2)
+        y = randint(1, self.board.width-2)
```

Comment:
this randint does not make a lot of sense to me, you just need to collect empty cells on the board as `positions`, and then just select a random one from that list

position = random.choices(positions)[0]